pennylane/ops/functions/iterative_qpe.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the qml.iterative_qpe function.
+"""
+
+import numpy as np
+import pennylane as qml
+
+
+def iterative_qpe(base, estimation_wire, iters):
+    r"""Performs the `iterative quantum phase estimation <https://arxiv.org/pdf/quant-ph/0610214.pdf>`_ circuit.
+
+    Given a unitary matrix :math:`U`, this function applies the circuit for iterative quantum phase
+    estimation and returns a list of the mid-circuit measurements.
+
+    Args:
+      base (Operator): the phase estimation unitary, specified as an :class:`~.Operator`
+      estimation_wire (Union[Wire, int, str]): the wire to be used for the estimation.
+      iters (int): the number of measurements to be performed
+
+    Returns:
+      list[MidMeasureMP]: the list of measurements performed
+
+    .. seealso:: :class:`~.QuantumPhaseEstimation`, :func:`~.measure`
+
+    **Example**
+
+    .. code-block:: python
+
+          dev = qml.device("default.qubit", shots = 5)
+
+          @qml.qnode(dev)
+          def circuit():
+
+              # Initial state
+              qml.PauliX(wires = [0])
+
+              # Iterative QPE
+              measurements = qml.iterative_qpe(qml.RZ(2., wires = [0]), estimation_wire = [1], iters = 3)
```

Comment:
The allowed types for estimation_wire don't include list? We should avoid it then in our examples
```suggestion
              measurements = qml.iterative_qpe(qml.RZ(2., wires = [0]), estimation_wire = 1, iters = 3)
```